### PR TITLE
#18141463 Feat: Social Login

### PR DESCRIPTION
**Pull request by rukundo-kevin**
# 1. Name of task
- Social authentication
- # 2. Pivot Tracker ID
-[ #18141463](https://www.pivotaltracker.com/story/show/18141463)

# 3. Branch Name
- `ft-social-login-18141463 `
# 4. Description of task
On the Login route, users should also be able to Login via Google or Facebook Button. Design of this functionality from a UI/UX perspective is up to us!

# Screenshots

Screen Shot 2022-06-07 at 13 24 09

# Testing

- Navigate to login and click on sign in with goole to login with your google account, remember to remove this line in `webpack.config.js` file, which enables https
`        port: 8080,
        https: true,`

- To test sign in with facebook, add this line in `webpack.config ` in devServer field to enable https in localhost as facebook requires it,
`        port: 8080,
        https: true,`

OR
use the vercel preview and test the deployed version

### CODE REVIEW

File: src/api/passwordApi.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import axios from 'axios';
 
-const apiUrl = 'https://z3a56d8ae-z32201c1c-gtw.z11b3bac6.rustrocks.cloud/';
+const apiUrl = 'http://127.0.0.1:7000/';
```
Comment (EmyRukundo) [inline src/api/passwordApi.js:3]: this URL should be in the .env file

File: test/root.test.js [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
 
 import App from '../src/components/app';
 
 test('on initial render, display something', () => {
   render(
-    <Provider store={store}>
       <App />
-    </Provider>
   );
 });
```
Comment (EmyRukundo) [inline test/root.test.js:10]: fix this

File: test/profile.test.js [hunk 3]
```diff
@@ -58,17 +54,5 @@ describe("User profile",()=>{
       fireEvent.change(screen.getByRole('departement'), {
         target: { value: '' },
       });
-      expect(screen.getByRole('UpdateButton')).toBeInTheDocument();
```
Comment (EmyRukundo) [inline test/profile.test.js:61]: why did you removed this test code?

**Review by EmyRukundo — CHANGES_REQUESTED** [review]

File: test/root.test.js (hunk 2 above)
Comment (rukundo-kevin) [reply test/root.test.js:10]: I have created a render method that wraps all components in redux store so this provider is no longer needed as this custom render already have the store.

**Review by NNesta — APPROVED** [review]
LGTM

**Review by SergeNm — APPROVED** [review]

**Review by mucyoalexaxel — APPROVED** [review]
Good Job @rukundo-kevin

**Pull request merged**